### ISSUE TITLE
Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)

### ISSUE THREAD

**Issue by giovannierre** [robertdebock/ansible-role-mysql#8]
## Describe the bug
When MariaDB is installed via *robertdebock.mysql* role, trying to use the 'mysql' client on the host machine (managed node) an error is raised:
```shell
root@dev2:~# mysql 
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)
```

This does not happen if MariaDB is installed directly.

## Playbook

```yaml
--- 
- hosts: dev2
  strategy: debug   
  become: yes
  become_user: root  
  roles:
    - role: robertdebock.bootstrap    
    - role: robertdebock.mysql   
      mysql_bind_address: 0.0.0.0     
      mysql_root_password: "mypassword" 
      mysql_databases:
        - name: prova_database
        # state:
        # target: 
        # encoding:
        # collation:     
      mysql_users:
        - name: admin
          password: another_password
          priv: "*.*:ALL"
          host: "%"
          update_password: on_create

```

## Output

The playbook works properly, no issue with it.

## Environment

- Control node OS: Ubuntu 20.04.1 LTS
- Control node Ansible version: 2.9.6
- Managed node OS: Ubuntu 20.04.1 LTS

## Workaround
Issue can be solved by making available the *mysqld.sock* file where it is expected, e.g. by creating a softlink:
```shell
~# ln -s /tmp/mysqld.sock /var/run/mysqld/mysqld.sock
```
or with a dedicated task in the playbook:
```yaml
post_tasks:
    - name: Create symbolic link for mysqld.sock to have it where it is expected      
      file:
        src: "/tmp/mysqld.sock"
        dest: "/var/run/mysqld/mysqld.sock"
        state: link
```
After the soft link is created, mysql client can connect without problems:
```shell
root@dev2:~# mysql
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 33
Server version: 10.3.25-MariaDB-0ubuntu0.20.04.1 Ubuntu 20.04

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MariaDB [(none)]>
```
So the solution works perfectly smooth but hopefully it can be solved also at role level.

**Comment by robertdebock** [comment]
I think this issue originates from an older version of the role, I remember fiddling with the socket file path. I'll confirm and get back to you.

Thanks for a great report on this issue.

**Comment by robertdebock** [comment]
Yes, this seems to be fixed in the current latest version (3.5.4)

## Preparation

```
image=ubuntu molecule test --destroy=never
```

## Testing
```
[robertdb@laptop ansible-role-mysql]$ image=ubuntu molecule login
root@mysql-ubuntu-latest:/# mysql
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 55
Server version: 10.3.25-MariaDB-0ubuntu0.20.04.1 Ubuntu 20.04

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MariaDB [(none)]> 
```

**Comment by giovannierre** [comment]
Thank you very much for the quick feedback!
Still not clear to me:
* if the *.my.cnf* file exposing password is really necessary or not (now yes, but with previous version of the role and *mysqld.sock* file position workaround apparantly not)
* why *mysql_socket* variable in the role is set to */var/run/mysqld/mysqld.sock* while the actual position is */tmp/mysqld.sock*
